Jenkinsfile

```diff
@@ -129,31 +131,68 @@ def getTests(String baseUrl) {
 }
 
 def getScenarios(String baseUrl, String testId){
-    def scenarios = [[Id: "-1", Name: "Default"]]
-    def isScenarioChoosen = "${params['Run with scenarios']}".toBoolean()
-    if (isScenarioChoosen) {
+    def scenarios = [[Id: "0", Name: "Default"]]
+    def isScenarioChoosen = "${params['Run with datasets']}".toBoolean()
+    def isScenarioPresentForTest = scenariosPresent(baseUrl, testId)
+    if (isScenarioChoosen && isScenarioPresentForTest) {
         def jsonSlurper = new JsonSlurper()
         URL url = new URL("${baseUrl}/projects/${params['Project']}/tests/${testId}/scenarios")
-        HttpURLConnection conn = url.openConnection()
-        conn.setRequestMethod("GET")
-        conn.setDoInput(true)
-        conn.setRequestProperty("Authorization", "APIKEY $useMangoApiKey")
-        conn.connect()
-        String content
-        def responseCode = conn.responseCode
-        if (responseCode == 200) {
-            InputStream inputStream = conn.getInputStream()
-            content = inputStream.getText()
-        } else {
-            throw new Exception("Scenarios get request failed with code: ${responseCode}")
-        }
+        String content = getWebResponseContent(url, "Scenarios")
         def scenarioPage = jsonSlurper.parseText(content)
         if (scenarioPage != null) {
             scenarioPage.each { scenario ->
-                echo "${scenario.Name}"        
+                echo "${scenario.Name}"
                 scenarios << [Id: scenario.Id, Name: scenario.Name]
             }
         }
+        return scenarios;
+    }
+    return null
+}
+
+def getAllScenarioIds(ArrayList<LinkedHashMap<String, String>> scenarioList) {
+    def listOfIds = []
+    scenarioList.each { item ->
+        listOfIds << item.Id
+    }
+    return listOfIds;
+}
+
+def addQueryParameterToUrl(String path, Map<String, Object> queryParams) {
```

Comment:
I think this method might be too generic in this context. It could instead be specific to preparing a test url; take a list of scenario ids and let this method take care of how it adds them to the url.

Reply:
I had made it generic since we have to provide for env. i picked the method from the other pr which is for environment.

![image](https://github.com/infuse-consulting/jenkins/assets/60608318/498c68cf-bea7-4df9-92ae-7216d96d6d7d)
i used it like this in the other PR
![image](https://github.com/infuse-consulting/jenkins/assets/60608318/5681541b-64c7-4a9c-8069-6f3f650b42ae)